abyfy_errors_test.go

```diff
@@ -0,0 +1,60 @@
+package httpapi
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+	mh "github.com/multiformats/go-multihash"
+)
+
+var randomSha256MH = mh.Multihash{0x12, 0x20, 0x88, 0x82, 0x73, 0x37, 0x7c, 0xc1, 0xc9, 0x96, 0xad, 0xee, 0xd, 0x26, 0x84, 0x2, 0xc9, 0xc9, 0x5c, 0xf9, 0x5c, 0x4d, 0x9b, 0xc3, 0x3f, 0xfb, 0x4a, 0xd8, 0xaf, 0x28, 0x6b, 0xca, 0x1a, 0xf2}
+
+func doAbyfyIpldNotFoundTest(t *testing.T, original error) {
+	originalMsg := original.Error()
+
+	rebuilt := abyfyIpldNotFoundFallbackToMSG(originalMsg)
+
+	rebuiltMsg := rebuilt.Error()
+
+	if originalMsg != rebuiltMsg {
+		t.Errorf("expected message to be %q; got %q", originalMsg, rebuiltMsg)
+	}
+
+	originalNotFound := ipld.IsNotFound(original)
+	rebuiltNotFound := ipld.IsNotFound(original)
+	if originalNotFound != rebuiltNotFound {
+		t.Errorf("expected Ipld.IsNotFound to be %t; got %t", originalNotFound, rebuiltNotFound)
+	}
+}
+
+func TestAbyfyIpldNotFound(t *testing.T) {
+	if err := abyfyIpldNotFoundFallbackToMSG(""); err != nil {
+		t.Errorf("expected empty string to give no error; got %T %q", err, err.Error())
+	}
+
+	for _, wrap := range [...]string{
+		"",
+		"merkledag: %w",
+		"testing: %w the test",
+		"%w is wrong",
+	} {
+		for _, err := range [...]error{
+			errors.New("file not found"),
+			errors.New(" not found"),
+			errors.New("Bad_CID not found"),
+			errors.New("network connection timeout"),
+			ipld.ErrNotFound{Cid: cid.Undef},
+			ipld.ErrNotFound{Cid: cid.NewCidV0(randomSha256MH)},
+			ipld.ErrNotFound{Cid: cid.NewCidV1(cid.Raw, randomSha256MH)},
+		} {
+			if wrap != "" {
+				err = fmt.Errorf(wrap, err)
+			}
+
+			doAbyfyIpldNotFoundTest(t, err)
```

Comment:
consider using t.Run() here so that each test case gets its own test

`t.Run(fmt.Sprintf("%s", err.Error()), func(t *testing.T) { doAbyfyIpldNotFoundTest(t, err) })`

Reply:
@guseggert I don't think it's a good idea, 
- it would be uselessly spammy, that test is running a matrix between the wrapping and the error so it would be `7*4 = 28` "sub tests"
- that tests runs in `0.00s` already adding that the test would spend more time creating subtests than actually running
- shouldn't fail often
- the error message will tell you expected vs got result already in the error message:
> ```go
> t.Errorf("expected message to be %q; got %q", originalMsg, rebuiltMsg)
> ```

Reply:
I don't feel strongly about it. I was reading this and had to build the input mentally, would be easier to just look at the test run.